src/compiler/corePublic.ts

```diff
@@ -114,25 +114,31 @@ namespace ts {
 
     /* @internal */
     namespace NativeCollections {
-        declare const Map: MapConstructor | undefined;
-        declare const Set: SetConstructor | undefined;
+        declare const self: any;
+
+        const globals = typeof globalThis !== "undefined" ? globalThis
+                  : typeof global !== "undefined" ? global
+                  : typeof self !== "undefined" ? self
+                  : undefined;
 
         /**
          * Returns the native Map implementation if it is available and compatible (i.e. supports iteration).
          */
         export function tryGetNativeMap(): MapConstructor | undefined {
             // Internet Explorer's Map doesn't support iteration, so don't use it.
+            const gMap = globals.Map;

```

Comment:
`globals` is possibly `undefined`.

```suggestion
            const gMap = globals?.Map;
```